File: AD-HOC/2322/main.rb
def missing(n, pieces)
  (0..n - 1).each do |i|
    return i + 1 if pieces[i] != i + 1
  end
  n
end

n = gets.to_i
pieces = gets.split.map(&:to_i).sort
puts missing(n, pieces)

File: AD-HOC/2322/Main.java
import java.util.Arrays;
import java.util.Scanner;
import java.util.stream.IntStream;

public class Main {

    static int missing(int n, int[] pieces) {
        Arrays.sort(pieces);
        for (int i = 0; i < n - 1; i++)
            if (pieces[i] != i + 1)
                return i + 1;
        return n;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] pieces = IntStream.range(0, n - 1).map(i -> scanner.nextInt()).toArray();
        System.out.println(missing(n, pieces));
        scanner.close();
    }
}
